pkg/asset/manifests/capiutils/helpers.go

```diff
@@ -0,0 +1,30 @@
+package capiutils
+
+import (
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/ipnet"
+)
+
+var (
+	defaultCIDR = ipnet.MustParseCIDR("10.0.0.0/16")
+)
+
+// CIDRFromInstallConfig generates the CIDR from the install config,
+// or returns the default CIDR if none is found.
+func CIDRFromInstallConfig(installConfig *installconfig.InstallConfig) *ipnet.IPNet {
+	if len(installConfig.Config.MachineNetwork) > 0 {
+		return &installConfig.Config.MachineNetwork[0].CIDR
+	}
+	return defaultCIDR
+}
+
+// IsEnabled returns true if the feature gate is enabled.
+func IsEnabled(installConfig *installconfig.InstallConfig) bool {
+	for _, enabledFeature := range installConfig.Config.FeatureGates {
+		if enabledFeature == string(v1.FeatureGateClusterAPIInstall) {
```

Comment:
v1.FeatureGateClusterAPIInstall is TechPreview so you also need to check whether the TechPreview feature set is enabled... Users can specify the TechPreview feature set and it enables all tech preview gates.